go-controller/pkg/ovn/pods_retry.go

```diff
@@ -0,0 +1,205 @@
+					}
+				} else {
+					klog.Infof("%s teardown retry failed; will try again later", podDesc)
+					podEntry.timeStamp = time.Now()
+					continue // if deletion failed we will not retry add
+				}
+			}
+			// check if we need to retry add
+			if retryAdd && podEntry.needsAdd {
+				klog.Infof("%s retry pod setup", podDesc)
+				if err := oc.ensurePod(nil, pod, true); err == nil {
+					klog.Infof("%s pod setup successful", podDesc)
+					delete(oc.retryPods, key) // this means retryDelete and retryAdd both worked, we can remove entry safely
+				} else {
+					klog.Infof("%s setup retry failed; will try again later", podDesc)
+					podEntry.timeStamp = time.Now()
+				}
+			}
+		} else {
+			klog.V(5).Infof("%s retry pod not after timer yet, time: %s", podDesc, podTimer)
+		}
+	}
+}
+
+// checkAndDeleteRetryPod deletes a specific entry from the map, if it existed, returns true
+func (oc *Controller) checkAndDeleteRetryPod(pod *kapi.Pod) bool {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	key := getPodNamespacedName(pod)
+	if _, ok := oc.retryPods[key]; ok {
+		delete(oc.retryPods, key)
+		return true
+	}
+	return false
+}
+
+// checkAndSkipRetryPod sets a specific entry from the map to be ignored for subsequent retries
+// if it existed, returns true
+func (oc *Controller) checkAndSkipRetryPod(pod *kapi.Pod) bool {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	key := getPodNamespacedName(pod)
+	if entry, ok := oc.retryPods[key]; ok {
+		entry.ignore = true
+		return true
+	}
+	return false
+}
+
+// unSkipRetryPod ensures a pod is no longer ignored for retry loop
+func (oc *Controller) unSkipRetryPod(pod *kapi.Pod) {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	key := getPodNamespacedName(pod)
+	if entry, ok := oc.retryPods[key]; ok {
+		entry.ignore = false
+	}
+}
+
+// initRetryAddPod tracks a pod that failed to be created to potentially retry later (needsAdd = true)
+// initially it is marked as skipped for retry loop (ignore = true)
+func (oc *Controller) initRetryAddPod(pod *kapi.Pod) {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	key := getPodNamespacedName(pod)
+	if entry, ok := oc.retryPods[key]; ok {
+		entry.timeStamp = time.Now()
+		entry.pod = pod
+		entry.needsAdd = true
+	} else {
+		oc.retryPods[key] = &retryEntry{pod, time.Now(), 1, true, true, nil}
+	}
+}
+
+// initRetryDelPod tracks a pod that failed to be deleted to potentially retry later (needsDel != nil)
+// initially it is marked as skipped for retry loop (ignore = true)
+func (oc *Controller) initRetryDelPod(pod *kapi.Pod) {
+	oc.retryPodsLock.Lock()
+	defer oc.retryPodsLock.Unlock()
+	key := getPodNamespacedName(pod)
+	if !util.PodWantsNetwork(pod) {
+		// create dummy logicalPortInfo for host-networked pods
+		mac, _ := net.ParseMAC("00:00:00:00:00:00")
+		oc.logicalPortCache.add("host-networked", getPodNamespacedName(pod),
```

Comment:
Why do you need this?

Reply:
just to have cleaner code, while retrying deletion of host-networked pods :( its cleaner to have a dummy port for using later on in the DeleteFunc since I moved the delete logic in removePod func similar to ensurePod.

Reply:
but deleteLogicalPort isn't called with host networked pods so I don't see a reason to add this to the logical port cache.

Reply:
lol, actually I don't know why I did this. I can't remember, I thought this was a brilliant idea for some reason ROFL. thanks for catching this, let me remove it.

Reply:
oh wait got it!

Reply:
@trozet : right so we want to retry a failed deletion of host-networked pods right? How do we know we want to retry? needsDel!=nil. And for host-networked pods it will be nil unless I add a dummy thing.
so I add a dummy value to portcache that gets used on L147 to fill needsDel which then gets deleted from the port cache in the DeleteFunc.

Reply:
its just a signal for retry when things go bad.

Reply:
let's not add cache entries and just create the lpinfo directly:
```
trozet@fedora ovn-kubernetes]$ git diff
diff --git a/go-controller/pkg/ovn/pods_retry.go b/go-controller/pkg/ovn/pods_retry.go
index 9d2639b18..aa8c3d1c5 100644
--- a/go-controller/pkg/ovn/pods_retry.go
+++ b/go-controller/pkg/ovn/pods_retry.go
@@ -138,13 +138,14 @@ func (oc *Controller) initRetryDelPod(pod *kapi.Pod) {
        oc.retryPodsLock.Lock()
        defer oc.retryPodsLock.Unlock()
        key := getPodNamespacedName(pod)
+       var portInfo *lpInfo
        if !util.PodWantsNetwork(pod) {
                // create dummy logicalPortInfo for host-networked pods
                mac, _ := net.ParseMAC("00:00:00:00:00:00")
-               oc.logicalPortCache.add("host-networked", getPodNamespacedName(pod),
-                       "host-networked", mac, []*net.IPNet{})
+               portInfo = &lpInfo{name: getPodNamespacedName(pod), mac: mac}
+       } else {
+               portInfo, _ = oc.logicalPortCache.get(key)
        }
-       portInfo, _ := oc.logicalPortCache.get(key)
        if entry, ok := oc.retryPods[key]; ok {
                entry.timeStamp = time.Now()
                entry.needsDel = portInfo

```
^ or something like this

Reply:
done.